src/js/actions/tools.js

```diff
@@ -695,37 +695,43 @@ define(function (require, exports) {
                 }
             };
 
-            var promise = this.transfer(shortcuts.addShortcut, activationKey, {}, activateTool);
-
-            promises.push(promise);
+            specs.push({
+                key: activationKey,
+                modifiers: {},
+                fn: activateTool
+            });
 
             // Add U as another shortcut for rectangle tool, hidden in here for now
             // FIXME: Change tool architecture to support multiple shortcuts for 1.1 - Barkin
             if (tool.id === "rectangle") {
-                var extraPromise = this.transfer(shortcuts.addShortcut, "U", {}, activateTool);
-                promises.push(extraPromise);
+                specs.push({
+                    key: "U",
+                    modifiers: {},
+                    fn: activateTool
+                });
             }
 
-            return promises;
+            return specs;
         }.bind(this), []);
     
         _vectorMaskHandler = function () {
             var toolStore = this.flux.store("tool"),
                 vectorMode = toolStore.getVectorMode() || false;
             
-            return this.transfer(changeVectorMaskMode, !vectorMode);
+            this.flux.actions.tools.changeVectorMaskMode(!vectorMode);
```

Comment:
This is an unrelated fix. It's not correct to transfer from an event handler like this because the locks are no longer being held. Instead, a new action should be enqueued.